apps/hash-frontend/src/pages/[shortname]/types/entity-type/[entity-type-id].page/definition-tab/use-editor-ontology-functions.ts

```diff
@@ -0,0 +1,86 @@
+import { EntityTypeEditorProps } from "@hashintel/type-editor";
+import { OwnedById } from "@local/hash-isomorphic-utils/types";
+import { useCallback } from "react";
+
+import { useBlockProtocolCreateEntityType } from "../../../../../../components/hooks/block-protocol-functions/ontology/use-block-protocol-create-entity-type";
+import { useBlockProtocolCreatePropertyType } from "../../../../../../components/hooks/block-protocol-functions/ontology/use-block-protocol-create-property-type";
+import { useBlockProtocolGetEntityType } from "../../../../../../components/hooks/block-protocol-functions/ontology/use-block-protocol-get-entity-type";
+import { useBlockProtocolGetPropertyType } from "../../../../../../components/hooks/block-protocol-functions/ontology/use-block-protocol-get-property-type";
+import { useBlockProtocolUpdateEntityType } from "../../../../../../components/hooks/block-protocol-functions/ontology/use-block-protocol-update-entity-type";
+import { useBlockProtocolUpdatePropertyType } from "../../../../../../components/hooks/block-protocol-functions/ontology/use-block-protocol-update-property-type";
+import { useFetchEntityTypes } from "../../../../../../shared/entity-types-context/hooks";
+import { useFetchLatestPropertyTypes } from "../latest-property-types-context";
+
+type OntologyFunctions = EntityTypeEditorProps["ontologyFunctions"];
+
+export const useEditorOntologyFunctions = (
+  ownedById: OwnedById | null,
+): EntityTypeEditorProps["ontologyFunctions"] => {
+  const { getEntityType } = useBlockProtocolGetEntityType();
+  const { createEntityType } = useBlockProtocolCreateEntityType(ownedById);
+  const { updateEntityType } = useBlockProtocolUpdateEntityType();
+
+  const { getPropertyType } = useBlockProtocolGetPropertyType();
+  const { createPropertyType } = useBlockProtocolCreatePropertyType(ownedById);
+  const { updatePropertyType } = useBlockProtocolUpdatePropertyType();
+
+  const refetchEntityTypes = useFetchEntityTypes();
+  const refetchPropertyTypes = useFetchLatestPropertyTypes();
+
+  const wrappedCreateEntityType = useCallback<
+    OntologyFunctions["createEntityType"]
+  >(
+    (args) => {
+      return createEntityType(args).then(async (res) => {
+        await refetchEntityTypes();
+        return res;
+      }) as any; // @todo fix this when types consistent
```

Comment:
This replicates what was previously happening for mutations from inside the type editgor – creating/updating, then awaiting the refetch, before moving on.